docs/build/guides/conventions/soroban-contract-init-template.mdx

```diff
@@ -0,0 +1,339 @@
+  />
+</head>
+
+This guide will walk you through the process of developing contract templates for working with Stellar smart contracts in frontend applications. This guide will cover the process of creating or developing the templates itself as well as how to work with the template to build, deploy, and generate smart contract bindings for the template.
+
+## Prerequisites
+
+Before you begin, ensure you have the following installed:
+
+1. [Rust](https://www.rust-lang.org/) and Cargo (for compiling smart contracts)
+2. [Node.js](https://nodejs.org/en) and npm (for running JavaScript deployment scripts)
+3. [Stellar CLI](../../smart-contracts/getting-started/setup.mdx#install-the-stellar-cli)
+
+## Anatomy of contract initialization (init) template
+
+Stellar smart contracts are written in Rust, but suppose we want our frontend web application to work with these contracts: this is where templates come into the equation. Templates allow us to work with smart contracts in our frontend application. At its core a template consists of a frontend library and a script for interacting with the Stellar CLI.
+
+Keep in mind you can use any frontend library to work with Stellar smart contracts. In this guide we'll use React to generate the template but it is worth noting that you can use whichever library you like or no library at all.
+
+We'll start off by installing the React library and then walking through creating the initialization script and finally wrap up with using the template to build, deploy, and generate contract bindings.
+
+## Creating the frontend
+
+The following code initializes a new Vite React project and installs the necessary dependencies:
+
+```bash
+npm create vite@latest soroban-react-template --template react
+cd soroban-react-template
+npm install
+npm run dev
+```
+
+## Creating the initialize script
+
+The initialization script is the heart of the contract template. This little but highly crucial script enables us to execute Stellar CLI commands which, in turn, enables us to build, deploy, and generate smart contract bindings.
+
+Usually, the initialization script is written in bash script but JavaScript can be used as well. This guide uses the latter but the same principles are involved when using bash scripts.
+
+We start off by creating `initialize.js` files in our frontend project directory.
+
+```bash
+touch initialize.js
+```
+
+This code creates the `initailze.js` file. Now let's open the file to begin writing the necessary code.
+
+### Installing the dependencies
+
+The following dependencies will simplify the process of interacting with files and directories which the `initialize.js` file does together with other functionalities.
+
+1. [glob](https://npmjs.com/package/glob)
+2. [dotenv](https:npmjs.com/package/dotenv)
+
+Install the above packages as dev dependencies by running the code below:
+
+```bash
```

Comment:
````suggestion
```bash npm2yarn
````

Just a suggestion, not a "must-have." You can add this to your `bash` code blocks when you are using `npm` or `yarn` commands and it'll provide the options for both, depending on what the user selects.

Reply:
I don't know for sure if it would work in the section above where you're create the project, installing, etc.